core/src/main/java/org/apache/iceberg/view/ViewMetadataParser.java

```diff
@@ -0,0 +1,213 @@
+ */
+package org.apache.iceberg.view;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.SchemaParser;
+import org.apache.iceberg.exceptions.RuntimeIOException;
+import org.apache.iceberg.io.InputFile;
+import org.apache.iceberg.io.OutputFile;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.util.JsonUtil;
+import org.apache.iceberg.util.PropertyUtil;
+
+class ViewMetadataParser {
+
+  // visible for testing
+  static final String FORMAT_VERSION = "format-version";
+  static final String LOCATION = "location";
+  static final String CURRENT_VERSION_ID = "current-version-id";
+  static final String VERSIONS = "versions";
+  static final String VERSION_LOG = "version-log";
+  static final String PROPERTIES = "properties";
+  static final String SCHEMAS = "schemas";
+  static final String CURRENT_SCHEMA_ID = "current-schema-id";
+
+  public static void overwrite(ViewMetadata metadata, OutputFile outputFile) {
+    internalWrite(metadata, outputFile, true);
+  }
+
+  public static void write(ViewMetadata metadata, OutputFile outputFile) {
+    internalWrite(metadata, outputFile, false);
+  }
+
+  public static void internalWrite(
+      ViewMetadata metadata, OutputFile outputFile, boolean overwrite) {
+    OutputStream stream = overwrite ? outputFile.createOrOverwrite() : outputFile.create();
+    try (OutputStreamWriter writer = new OutputStreamWriter(stream, StandardCharsets.UTF_8)) {
+      JsonGenerator generator = JsonUtil.factory().createGenerator(writer);
+      generator.useDefaultPrettyPrinter();
+      toJson(metadata, generator);
+      generator.flush();
+    } catch (IOException e) {
+      throw new RuntimeIOException(e, "Failed to write json to file: %s", outputFile);
+    }
+  }
+
+  public static void toJson(ViewMetadata metadata, JsonGenerator generator) throws IOException {
+    generator.writeStartObject();
+
+    generator.writeNumberField(FORMAT_VERSION, metadata.formatVersion());
+    generator.writeStringField(LOCATION, metadata.location());
+
+    JsonUtil.writeStringMap(PROPERTIES, metadata.properties(), generator);
+
+    if (metadata.schemas() != null && !metadata.schemas().isEmpty()) {
+      generator.writeNumberField(CURRENT_SCHEMA_ID, metadata.currentSchemaId());
+      generator.writeArrayFieldStart(SCHEMAS);
+      for (Schema schema : metadata.schemas()) {
+        SchemaParser.toJson(schema, generator);
+      }
+
+      generator.writeEndArray();
+    }
```

Comment:
I need to think about this more. I don't like the idea of schemas being optional in the spec. If an engine chooses to dynamically construct at read time and ignore the schema, that's their choice. I think at minimum when updating the view there must be a schema that is set; the metadata should take some stance on what the output Iceberg schema for the view is, otherwise it seems "under"-spec'd and will open the gates to more interpretation than we would probably like. @rdblue @jackye1995 thoughts?

Reply:
I remember there is a use case in Netflix that requires it to be optional that we discussed in a previous PR, do you remember that?

Reply:
I found https://github.com/apache/iceberg/pull/3188#discussion_r748303084 this thread on the original spec proposal, and then for view representation case there was https://github.com/apache/iceberg/pull/4925#discussion_r1015649887. 

So based on these two I don't think it's required that it's optional? I think it's just a convenience to avoid an engine having to pass in a schema at the time of view creation. Based on the second link it seems there is some complexity related to aliasing but if we store the pre-aliased schema that should be sufficient, the post alias columns can be subsequently derived.

Schema handling will probably be a focal point around the view spec so it's important we get this right as we try and conclude this implementation. Right now schemaID in `SQLViewRepresentation` is optional. Maybe we revisit this, and make schemaID at the version level, since I think all representations should have the same Iceberg schema. 

Would love to get others thoughts here.

Reply:
I think it is optional for convenience. But I don't think that's a very strong reason.

It would be strange for a view to not have a schema when you inspect it. And it would similarly be unexpected for a view to be created directly by a human rather than an engine. I think it makes sense to require the `schemas`.

However, I would also make a couple of other changes. First, `current-schema-id` doesn't make sense because it would need to be maintained whenever the view version changes. Why keep it in sync when we could just drop the field and use the current version's schema?

Also, I don't think that the view version tracks a schema right now, but since we expect all representations to produce the same schema, I think that it should. I'd probably move `schema-id` from the SQL representation to the view version.

Reply:
Totally agreed @rdblue, I'll work on a PR to move the schema to ViewVersion.